### DISCUSSION TITLE
Check for Auth::user() inside jobs and other queueables

### DISCUSSION THREAD

**Discussion by danielbrodin** [enlightn/enlightn#52] [Ideas]
**Is your feature request related to a problem? Please describe.**
Auth::user() can't be used inside queueables since there is no user then, but that doesn't stop from using it by accident. This is something that from my knowledge can't be found in tests either since they are run sync and have an auth user.

**Describe the solution you'd like**
A warning about using Auth::user / auth()->user() inside queueables, at least if the QUEUE_CONNECTION isn't sync.

**Comment by paras-malhotra** [comment]
Hi @danielbrodin, thanks for posting the feature request. I'm not quite sure if this would be foolproof though. There may be custom auth guards configured in the app that authenticate via a different way, e.g. perhaps a token provided to the queued job is intended to authenticate the user (for say stateless apps).

Would love to hear some feedback from other folks on this, so leaving this open for a while.

**Comment by danielbrodin** [comment]
I have looked for a way to use the auth user inside a queued job, but haven't found any (except passing it when creating it of course). But that isn't to say that there is no way of course :)
